lms/djangoapps/certificates/apis/v0/tests/test_views.py

```diff
@@ -0,0 +1,108 @@
+"""
+Tests for the Certificate REST APIs.
+"""
+from django.core.urlresolvers import reverse
+from rest_framework import status
+from rest_framework.test import APITestCase
+
+from certificates.tests.factories import GeneratedCertificateFactory
+from certificates.models import CertificateStatuses
+from course_modes.models import CourseMode
+
+from student.tests.factories import UserFactory
+from xmodule.modulestore.tests.django_utils import SharedModuleStoreTestCase
+from xmodule.modulestore.tests.factories import CourseFactory
+
+
+class CertificatesRestApiTest(SharedModuleStoreTestCase, APITestCase):
+    """
+    Test for the Certificates REST APIs
+    """
+    @classmethod
+    def setUpClass(cls):
+        super(CertificatesRestApiTest, cls).setUpClass()
+        cls.course = CourseFactory.create(
+            org='edx',
+            number='verified',
+            display_name='Verified Course'
+        )
+
+    def setUp(self):
+        super(CertificatesRestApiTest, self).setUp()
+
+        self.student = UserFactory.create(password='test')
+        self.student_no_cert = UserFactory.create(password='test')
+
+        GeneratedCertificateFactory.create(
+            user=self.student,
+            course_id=self.course.id,
+            status=CertificateStatuses.downloadable,
+            mode='verified',
+            download_url='www.google.com',
+            grade="0.88"
+        )
+
+        self.namespaced_url = 'certificates_api:v0:certificates:detail'
+
+    def get_url(self, username):
+        """
+        Helper function to create the url for certificates
+        """
+        return reverse(
+            self.namespaced_url,
+            kwargs={
+                'course_id': self.course.id,
+                'username': username
+            }
+        )
+
+    def test_permissions(self):
+        """
+        Test that only the owner of the certificate can access the url
+        """
+        # anonymous user
+        resp = self.client.get(self.get_url(self.student.username))
+        self.assertEqual(resp.status_code, status.HTTP_401_UNAUTHORIZED)
+
+        # another student
+        self.client.login(username=self.student_no_cert.username, password='test')
+        resp = self.client.get(self.get_url(self.student.username))
+        # gets 404 and 403 for security reasons
```

Comment:
i did not write the comment right:
will change to `# gets 404 instead of 403 for security reasons`